apps/web/src/components/admin/UserGuideRow.tsx

```diff
@@ -1,28 +1,146 @@
 import dayjs from 'dayjs';
+import { useEffect, useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faReply, faSpinner, faTrash } from '@fortawesome/free-solid-svg-icons';
+import { mutate } from 'swr';
 import { UserGuide } from '@ien/common';
-import { downloadGuide } from '../../services/admin';
+import { Disclosure } from '@components';
+import {
+  deleteUserGuide,
+  downloadUserGuide,
+  getUserGuideVersions,
+  restoreUserGuide,
+} from '../../services/admin';
 
 interface UserGuideProps {
   file: UserGuide;
+  showVersions: boolean;
 }
 
-export const UserGuideRow = ({ file: { name, lastModified } }: UserGuideProps) => {
-  const open = async () => {
-    const url = await downloadGuide(name);
-    window.open(url);
+export const UserGuideRow = ({ file, showVersions }: UserGuideProps) => {
+  const [versions, setVersions] = useState<UserGuide[]>();
+  const [loading, setLoading] = useState(false);
+  const [selectedVersion, setSelectedVersion] = useState<string | undefined>();
+
+  const getVersions = (open: boolean) => {
+    if (open) {
```

Comment:
whats the purpose of the `boolean`?

Reply:
If the disclosure is expanded, then `open` is true and we need to query versions to display them.